distributed/client.py

```diff
@@ -3944,13 +3937,23 @@ def dump_cluster_state(
                     from yaml import Loader
                 with open("filename") as fd:
                     state = yaml.load(fd, Loader=Loader)
+        **kwargs:
+            Any additional arguments to :func:`fsspec.open` when writing to a URL.
+            Ignored when writing to a local file.
 
         """
+        filename = str(filename)
+        method = (
+            self._dump_cluster_state_remote
+            if "://" in filename
```

Comment:
I'm probably overthinking this, but there are valid `fsspec` urls with `://` in them that are targeting the local filesystem (specifically things around caching and zipping data). I'm not sure if there are any good use-cases for that, but there may be a more reliable check along the lines of
```python
fs, url = fsspec.core.url_to_fs(filename)
method = (
    self._dump_cluster_state_local
    if fs.local_file
    else self._dump_cluster_state_remote
)
```

This would also allow you to use `fsspec` on the client side as well (again, not sure if there are any good use-cases, but would be a bit more consistent)

Reply:
Yeah, I thought about this a bit. I'm kinda leaning towards the current dumb check for `://` because:
* It's easy to document. The fsspec docs don't say much about what's considered a local file or not (for example, `simplecache::s3://foo` is local, but `s3://foo` isn't).
* I don't think any of the local filesystems (zip, tar, git, etc.) actually support writes. So in practice, I don't think you can give `tar://dumps.tar.gz/dump` and write to it. `file://dump` is the only one I've found that does, which is just a more inconvenient way of writing `dump`, which I doubt anyone would do.
* In the rare case that your scheduler has s3fs installed, but your client doesn't, checking `url_to_fs("s3://foo")` would raise an error

I am going to add an explicit `write_locally` flag though, so you can override this inference if you want. Could be useful in cases where your scheduler is running on NFS, or your client has access to a bucket that the cluster doesn't, etc.

Reply:
Yeah, that's fair. I do wish `fsspec` provided some better utilities for reasoning about URLs. I'm mostly reacting against every python project having it's own slightly-mutually-incompatible set of checks for stuff like this. `fsspec` is well-positioned to have a battery of helper functions, but currently does not.

>  I am going to add an explicit write_locally flag though, so you can override this inference if you want.

This sounds like a nice compromise